cli/azd/docs/guiding-principles.md

```diff
@@ -0,0 +1,109 @@
+
+- Purpose: Display current state, configuration, and system information
+- Examples: `list`, `show`, `version`
+- Principle: Read-only operations that help users understand current state
+
+#### **Edit** (Configuration & Management)
+
+- Purpose: Modify configuration, manage resources, and setup operations
+- Examples: `set`, `reset`, `unset`, `config`, `refresh`, `select`, `add`, `remove`, `install`, `uninstall`, `new`
+- Principle: Operations that change system state or configuration
+
+#### **Run** (Actions & Operations)
+
+- Purpose: Execute workflows, deploy resources, and perform active operations
+- Examples: `up`, `down`, `init`, `package`, `provision`, `deploy`, `run`
+- Principle: Operations that perform work or execute processes
+
+#### **Other** (Specialized Operations)
+
+- Purpose: Specialized tasks that don't fit standard categories
+- Examples: `restore`, `monitor`, `generate`, `login`, `logout`
+- Principle: Unique operations specific to particular workflows
+
+## Design Guidelines for Extensions
+
+### 1. **Command Integration Strategy**
+
+- New functionality should extend existing command categories
+- Use a verb-first structure, where the primary action (e.g., add, create, delete, list) is the top-level command, and the target entity or context follows as an argument or subcommand.
+- Example: `azd add <new-resource-type>` instead of `azd <new-resource-type> add`
+
+### 2. **Parameter and Flag Consistency**
+
+- Reuse established parameter patterns across new commands
+- Maintain consistent naming conventions (e.g., `--subscription`, `--name`, `--type`)
+- Provide sensible defaults to reduce cognitive load
+
+### 3. **Help and Discoverability**
+
+- Integrate new functionality into existing `azd help` structure
+- Provide contextual guidance within established command flows
+- Maintain documentation consistency across core and extended features
+
+### 4. **Template and Resource Integration**
+
+- Leverage existing template system for new resource types
+- Follow established template discovery and management patterns
+- Integrate new resources into azd's resource lifecycle management
+
+## Implementation Benefits
+
+- **User Familiarity**: Builds on known command patterns and reduces learning curve
+- **Discoverability**: New capabilities are found through existing workflows
+- **Consistency**: Predictable behavior across all azd commands and extensions
+- **Maintainability**: Systematic approach reduces complexity and technical debt
+- **Extensibility**: Clear framework for adding capabilities without breaking existing patterns
+- **Ecosystem Growth**: Provides foundation for third-party extensions and integrations
+
+## Future Considerations
+
+This framework enables:
+
+- Advanced workflow automation
+- Enhanced developer productivity features
+- Consistent user experience across all `azd` functionality
+- Integration of new Azure services and capabilities
+- Third-party extension development
```

Comment:
I think in a month or two we should update these to add a section on "Design as an Extension first" to push teams to use the extension framework rather than trying to build in azd core.  I'd like to make a little more progress knocking down sharp edges first though.  Added to https://github.com/Azure/azure-dev/issues/5767 to track.